sdk/identity/src/token_credentials/azureauth_cli_credentials.rs

```diff
@@ -0,0 +1,185 @@
+use azure_core::{
+    auth::{AccessToken, TokenCredential, TokenResponse},
+    error::{Error, ErrorKind, ResultExt},
+};
+use oauth2::ClientId;
+use serde::Deserialize;
+use std::process::Command;
+use std::str;
+use time::OffsetDateTime;
+
+mod unix_date_string {
+    use azure_core::error::{Error, ErrorKind};
+    use serde::{Deserialize, Deserializer};
+    use time::OffsetDateTime;
+
+    pub fn parse(s: &str) -> azure_core::Result<OffsetDateTime> {
+        let as_i64 = s.parse().map_err(|_| {
+            Error::with_message(ErrorKind::DataConversion, || {
+                format!("unable to parse expiration_date '{s}")
+            })
+        })?;
+
+        OffsetDateTime::from_unix_timestamp(as_i64).map_err(|_| {
+            Error::with_message(ErrorKind::DataConversion, || {
+                format!("unable to parse expiration_date '{s}")
+            })
+        })
+    }
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<OffsetDateTime, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        parse(&s).map_err(serde::de::Error::custom)
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+struct CliTokenResponse {
+    pub token: AccessToken,
+    #[serde(with = "unix_date_string")]
+    pub expiration_date: OffsetDateTime,
+}
+
+/// Authentication Mode
+///
+/// Note: While the azureauth CLI supports devicecode, users wishing to use
+/// devicecode should use `azure_identity::device_code_flow`
+#[derive(Debug, Clone, Copy)]
+pub enum AzureauthCliMode {
+    All,
+    #[cfg(target_os = "windows")]
+    IntegratedWindowsAuth,
+    Broker,
```

Comment:
```suggestion
    #[cfg(target_os = "windows")]
    Broker,
```
Broker would also need it's own `cfg` atribute right?

Reply:
@demoray ping.